### PR TITLE
Add projects from `root` stage. Add `account` project

### PR DESCRIPTION
**Pull request by aknysh**
## what
* Add projects (`dns` and `iam`) from `root` stage (and  eliminate dependencies of those projects on `terraform-account-metadata`)
* Add `accounts` project

## why
* Keep all reference architecture modules in one repo
* Provision member accounts (`prod`, `staging`, `dev`, `audit`) in an organization

### CODE REVIEW

File: aws/account/terraform.tfvars.example [hunk 1]
```diff
@@ -0,0 +1,11 @@
+prod_account_name="prod"
+prod_account_email="info+prod@cloudposse.co"
+
+staging_account_name="staging"
```
Comment (osterman) [inline aws/account/terraform.tfvars.example]: I think we should maybe separate them into N files.

image

File: aws/dns/stages.tf [hunk 2]
```diff
@@ -0,0 +1,22 @@
+resource "aws_route53_zone" "dns_zone" {
```
Comment (osterman) [inline aws/dns/stages.tf]: Why is this file called `stages.tf`?

File: aws/iam/audit.tf [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  description = "Audit account ID"
+}
+
+# Provision group access to audit account. Careful! Very few people, if any should have access to this account.
+module "organization_access_group_audit" {
+  source            = "git::https://github.com/cloudposse/terraform-aws-organization-access-group.git?ref=tags/0.1.2"
+  namespace         = "${var.namespace}"
+  stage             = "audit"
+  name              = "admin"
+  user_names        = ["erik", "andriy", "igor", "sarkis"]
```
Comment (osterman) [inline aws/iam/audit.tf]: Let's switch to email addresses. Move to `terraform.tfvars`

File: aws/iam/prod.tf [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  description = "Prod account ID"
+}
+
+# Provision group access to production account
+module "organization_access_group_prod" {
+  source            = "git::https://github.com/cloudposse/terraform-aws-organization-access-group.git?ref=tags/0.1.2"
+  namespace         = "${var.namespace}"
+  stage             = "prod"
+  name              = "admin"
+  user_names        = ["erik", "andriy", "igor", "sarkis"]
```
Comment (osterman) [inline aws/iam/prod.tf]: move to var file

File: aws/iam/root.tf [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# Provision group access to root account with MFA
+module "organization_access_group_root" {
+  source           = "git::https://github.com/cloudposse/terraform-aws-iam-assumed-roles.git?ref=tags/0.2.0"
+  namespace        = "${var.namespace}"
+  stage            = "root"
+  admin_name       = "admin"
+  readonly_name    = "readonly"
+  admin_user_names = ["erik", "andriy", "igor", "sarkis"]
```
Comment (osterman) [inline aws/iam/root.tf]: move to var file

File: aws/dns/stages.tf (hunk 2 above)
Comment (aknysh) [reply aws/dns/stages.tf]: the file will be copied into the stages (prod, staging, dev) in `Dockerfiles`.
the file is re-used.
not sure what to call it better.

Comment (aknysh) [reply aws/dns/stages.tf]: maybe just `dns`?

File: aws/account/audit.tf [hunk 6]
```diff
@@ -0,0 +1,21 @@
+variable "audit_account_name" {
+  type        = "string"
+  description = "Audit account name"
+  default     = "audit"
+}
+
```
Comment (osterman) [inline aws/account/audit.tf]: Thinking `account/` should be `accounts/`

File: aws/dns/stages.tf (hunk 2 above)
Comment (osterman) [reply aws/dns/stages.tf]: Ok, I think perhaps we are overloading `dns/`. 

Thinking we should have `root-dns` and maybe `account-dns`

File: aws/account/main.tf [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+variable "account_role_name" {
+  type        = "string"
+  description = "IAM role that Organizations automatically preconfigures in the new member account"
+  default     = "OrganizationAccountAccessRole"
+}
+
+variable "account_iam_user_access_to_billing" {
+  type        = "string"
+  description = "If set to ALLOW, the new account enables IAM users to access account billing information if they have the required permissions. If set to DENY, then only the root user of the new account can access account billing information"
```
Comment (osterman) [inline aws/account/main.tf]: Use markdown code? 
```
If set to `ALLOW`
```

File: aws/dns/stages.tf (hunk 2 above)
Comment (osterman) [reply aws/dns/stages.tf]: (open to alternative names too... but i think we should create more folders rather than less)

Comment (osterman) [reply aws/dns/stages.tf]: A given folder should never serve 2 purposes.

**Pull request merged**